addons/event/models/event_registration.py

```diff
@@ -308,10 +308,15 @@ def _update_mail_schedulers(self):
             return
 
         onsubscribe_schedulers.update({'mail_done': False})
-        # we could simply call _create_missing_mail_registrations and let cron do their job
-        # but it currently leads to several delays. We therefore call execute until
-        # cron triggers are correctly used
-        onsubscribe_schedulers.with_user(SUPERUSER_ID).execute()
+        # either trigger the cron, either run schedulers immediately (scaling choice)
+        async_scheduler = self.env['ir.config_parameter'].sudo().get_param('event.event_mail_async')
+        if async_scheduler:
+            self.env.ref('event.event_mail_scheduler')._trigger()
```

Comment:
not 100% sure what will happen if you have 2 registrations at the same time, will the cron be called 2 times ?

Reply:
It is going to create 2 triggers, but as far as I know cron will be executed only once to "consume" all triggers before its execution time, remembering discussion with @Julien00859  about it. Can ping him for confirmation.

Reply:
> cron will be executed only once to "consume" all triggers before its execution time

Indeed